src/Models/ContainerDeviceBase.ts

```diff
@@ -130,46 +127,47 @@ export abstract class ContainerDeviceBase implements Device {
       }
     }
 
-    // load project config
-    const configPath = path.join(
-        this.projectFolder, FileNames.vscodeSettingsFolderName,
-        constants.configFile);
-    if (!await FileUtility.fileExists(ScaffoldType.Workspace, configPath)) {
-      const message = `Config file does not exist. Please check your settings.`;
-      await vscode.window.showWarningMessage(message);
+    const tasks = await vscode.tasks.fetchTasks();
+    if (!tasks || tasks.length < 1) {
       return false;
     }
 
-    const fileContent =
-        await FileUtility.readFile(ScaffoldType.Workspace, configPath);
-    const config: Config = JSON.parse(fileContent as string);
+    const compileTask = tasks.filter(task => {
+      return task.name === constants.compileTaskName;
+    });
+    if (!compileTask || compileTask.length < 1) {
+      return false;
+    }
 
-    channelShowAndAppendLine(
-        this.channel, `Compiling ${this.name} device code...`);
     try {
-      await runCommand(
-          config.buildCommand, [], this.projectFolder, this.channel);
+      await vscode.tasks.executeTask(compileTask[0]);
     } catch (error) {
-      throw new Error(`Failed to compile ${
-          this.name} device code. Error message: ${error.message}`);
+      throw new Error(`Failed to execute compilation task.`);
     }
 
-    // If successfully compiled, copy compiled files to user workspace
-    if (await FileUtility.directoryExists(
-            ScaffoldType.Workspace, config.buildTarget)) {
-      const getOutputFileCmd =
-          `cp -rf ${config.buildTarget} ${this.outputPath}`;
-      try {
-        await runCommand(getOutputFileCmd, [], '', this.channel);
-      } catch (error) {
-        throw new Error(`Failed to copy compiled files to output folder ${
-            this.outputPath}. Error message: ${error.message}`);
+    vscode.tasks.onDidEndTaskProcess(async (event) => {
+      if (event.exitCode === 0) {
+        // If task is successfully executed, copy compiled files to user
+        // workspace
+        if (await FileUtility.directoryExists(
+                ScaffoldType.Workspace, constants.outputPathInContainer)) {
+          const getOutputFileCmd =
+              `cp -rf ${constants.outputPathInContainer}/* ${this.outputPath}`;
+          try {
+            await runCommand(getOutputFileCmd, [], '', this.channel);
+          } catch (error) {
+            throw new Error(`Failed to copy compiled files to output folder ${
+                this.outputPath}. Error message: ${error.message}`);
+          }
+          return true;
+        } else {
+          throw new Error(
```

Comment:
since we throw exception, do we still need return true/false to label result?

Reply:
Can be removed. Fixed.